dashboard/modules/serve/serve_agent.py

```diff
@@ -53,7 +53,13 @@ async def get_all_deployments(self, req: Request) -> Response:
         if client is None:
             config = ServeApplicationSchema.get_empty_schema_dict()
         else:
-            config = client.get_app_config()
+            try:
+                config = client.get_app_config()
+            except ray.exceptions.RayTaskError:
+                # Task failure sometimes are due to GCS
+                # failure. When GCS failed, we expect a longer time
+                # to recover.
+                config = ServeApplicationSchema.get_empty_schema_dict()
```

Comment:
hmm I think we should raise an exception here instead of returning the empty schema -- that is incorrect behavior (there may actually be an active config)

cc @shrekris-anyscale

Reply:
Agreed– the empty schema indicates that there's no active config. We need to communicate that the GCS is currently recovering in this exception.

Reply:
@edoakes @shrekris-anyscale   I need to sync with @brucez-anyscale about what serve operator wants for this case. I'll update it and the test early in the afternoon.

Reply:
Ok sounds good. I think the most intuitive behavior would be to return a `503 Service Unavailable` in these cases.